src/containerapp/azext_containerapp/_cloud_build_utils.py

```diff
@@ -0,0 +1,213 @@
+
+    try:
+        done_spinner = False
+        fail_spinner = False
+        cancel_spinner = False
+        font_bold_green = "\033[32;1m"
+        font_bold_red = "\033[31;1m"
+        font_bold_yellow = "\033[33;1m"
+        font_default = "\033[0m"
+        font_bold = "\033[0;1m"
+        font_blue = "\033[94m"
+        substatus_indentation = "              "
+        def display_spinner(task_title):
+            # Hide the cursor
+            print('\033[?25l', end="")
+            def spin():
+                loop_counter = 0
+                start_time = time.time()
+                while (done_spinner == False and fail_spinner == False and cancel_spinner == False):
+                    loop_counter = (loop_counter + 1) % 17
+                    loading_bar_left_spaces_count =  loop_counter - 9 if loop_counter > 9 else 0
+                    loading_bar_right_spaces_count =  6 - loop_counter if loop_counter < 7 else 0
+                    spinner = f"[{' ' * loading_bar_left_spaces_count}{'=' * (7 - loading_bar_left_spaces_count - loading_bar_right_spaces_count)}{' ' * loading_bar_right_spaces_count}]"
+                    time_elapsed = time.time() - start_time
+                    print(f"\r    {spinner} {task_title} ({time_elapsed:.1f}s)", end="", flush=True)
+                    time.sleep(0.15)
+                if (cancel_spinner):
+                    print(f"\r    {font_bold_yellow}Canceled:{font_default} {task_title} ({time_elapsed:.1f}s)")
+                    log_in_file(f"Canceled: {task_title} ({time_elapsed:.1f}s)", logs_file, no_print=True)
+                elif (fail_spinner):
+                    print(f"\r    {font_bold_red}(X) Fail:{font_default} {task_title} ({time_elapsed:.1f}s)")
+                    log_in_file(f"Fail: {task_title} ({time_elapsed:.1f}s)", logs_file, no_print=True)
+                else:
+                    print(f"\r    {font_bold_green}(✓) Done:{font_default} {task_title} ({time_elapsed:.1f}s)")
+                    log_in_file(f"Done: {task_title} ({time_elapsed:.1f}s)", logs_file, no_print=True)
+                # Display the cursor
+                print('\033[?25h', end="")
+            thread = Thread(target=spin)
+            thread.start()
+            return thread
+
+        log_in_file(f"\n  {font_bold}Preparing the Container Apps Cloud Build environment{font_default}\n", logs_file)
+
+        # List the builders in the resource group
+        thread = display_spinner("Listing the builders available in the Container Apps environment")
+        builders_list_json_content = BuilderClient.list(cmd, resource_group_name)
+        done_spinner = True
+        thread.join()
+
+        if len(builders_list_json_content["value"]) > 0:
+            builder_name = builders_list_json_content["value"][0]["name"]
+            log_in_file(f"{substatus_indentation}Builder selected: {builder_name}", logs_file)
```

Comment:
Do we not allow more granularity when it comes to selecting the builder? Specifically, if we are given the `environment_name` argument to this `run_cloud_build` function, can we filter the results of `BuilderClient.list` by those associated with the given environment and then select the first one?

Reply:
Yes, you're right as well.